vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceOverviewController.java

```diff
@@ -0,0 +1,317 @@
+        Map<String, List<ModuleLinkDisplay>> spaceToModuleLinksMap = new HashMap<>();
+        getSpaceToAllOtherLinks(spaceToModuleLinks, spaceToModuleLinksMap, true);
+
+        Iterable<SpaceLinkDisplay> spaceToSpaceLinksList = spaceLinkDisplayRepository.findAll();
+        Map<String, List<SpaceLinkDisplay>> spaceToSpaceLinksMap = new HashMap<>();
+        getSpaceToAllOtherLinks(spaceToSpaceLinksList, spaceToSpaceLinksMap, false);
+
+        ObjectMapper mapper = new ObjectMapper();
+        ArrayNode nodeArray = mapper.createArrayNode();
+        ArrayNode linkArray = mapper.createArrayNode();
+
+        Iterable<Space> allSpacesList = spaceRepo.findAll();
+        List<String> allNodeList = new ArrayList<>();
+        getSpaceLinkMap(request.getContextPath(), allSpacesList, mapper, nodeArray, allNodeList, spaceLinkMap,
+                spaceToModuleLinksMap, spaceToSpaceLinksMap);
+
+        Iterable<Module> allModulesList = moduleRepo.findAll();
+        createNodeForOverviewGraph(request.getContextPath(), allModulesList, mapper, nodeArray, allNodeList, true);
+
+        createEdgeForOverviewGraph(mapper, linkArray, allNodeList, spaceLinkMap);
+        model.addAttribute("overviewNode", mapper.writeValueAsString(nodeArray));
+        model.addAttribute("overviewLink", mapper.writeValueAsString(linkArray));
+        return "staff/spaces/graph";
+    }
+
+    /**
+     * To retrieve all space to space and space to module links corresponding to
+     * each space
+     * 
+     * @param contextPath           The contextpath of the application is being
+     *                              passed
+     * @param allSpacesList         List of all spaces which are available in vspace
+     *                              is being passed
+     * @param mapper                Passing this object to create object node for
+     *                              creating json of space and module node in the
+     *                              spaceoverview graph
+     * @param nodeArray             To create the json for each node of
+     *                              spaceoverview graph
+     * @param allNodesList          List of all the nodes of spaceoverview graph
+     * @param spaceLinkMap          For holding list of all links corresponding to
+     *                              each space
+     * @param spaceToModuleLinksMap For getting list of links from one space to
+     *                              other modules
+     * @param spaceToSpaceLinksMap  For getting list of links from one space to
+     *                              other spaces
+     */
+    private void getSpaceLinkMap(String contextPath, Iterable<Space> allSpacesList, ObjectMapper mapper,
+            ArrayNode nodeArray, List<String> allNodesList, Map<String, List<String>> spaceLinkMap,
+            Map<String, List<ModuleLinkDisplay>> spaceToModuleLinksMap,
+            Map<String, List<SpaceLinkDisplay>> spaceToSpaceLinksMap) {
+        if (allSpacesList != null) {
```

Comment:
this check should be where the method is being called

Reply:
This check is necessary as immediately after that there is a loop on it.